### PR TITLE
TD Heuristic for tests mentioned in PR body, less verbose TD printing

### PR DESCRIPTION
**Pull request by clee2000**
Move tests that are mentioned in PR body or commit message to front.  Also attempts to find any issues/PRs mentioned in the PR body and search for those too (ex if you link a disable issue and that issue contains the test file that it was failing on)

looking for: dynamo/test_export_mutations

Also removes some printed information in TD

### CODE REVIEW

File: .github/workflows/trunk.yml [hunk 1]
```diff
@@ -27,192 +27,192 @@ jobs:
       contents: read
 
   # Build PyTorch with BUILD_CAFFE2=ON
-  caffe2-linux-jammy-py3_8-gcc11-build:
```
Comment (clee2000) [inline .github/workflows/trunk.yml:30]: Remember to change this back

File: .github/workflows/target_determination.yml [hunk 2]
```diff
@@ -39,6 +39,7 @@ jobs:
         id: td
         continue-on-error: true
         env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (osalpekar) [inline .github/workflows/target_determination.yml:42]: Is this alright? cc @atalman

File: tools/testing/target_determination/heuristics/mentioned_in_pr.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+from tools.testing.target_determination.heuristics.interface import (
+    HeuristicInterface,
+    TestPrioritizations,
+)
+from tools.testing.test_run import TestRun
+
+GITHUB_TOKEN = os.environ.get("GITHUB_TOKEN", "")
+
+
+# This heuristic gives a test a rating of 1 if it is mentioned in the PR or a
```
Comment (osalpekar) [inline tools/testing/target_determination/heuristics/mentioned_in_pr.py:17]: also searching in issue body for issues linked in PR title?

File: tools/testing/target_determination/heuristics/mentioned_in_pr.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+                test in commit_messages
+                or test in pr_body
+                or any(test in body for body in linked_issue_bodies)
+            ):
+                mentioned.append(test)
+
+        return TestPrioritizations(tests, {TestRun(test): 1 for test in mentioned})
+
+
+def get_git_commit_info() -> str:
```
Comment (osalpekar) [inline tools/testing/target_determination/heuristics/mentioned_in_pr.py:67]: do we not have this in some sort of utils file across the different heuristics?

File: tools/testing/target_determination/heuristics/mentioned_in_pr.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    return (
+        subprocess.check_output(
+            ["git", "log", f"{base_commit}..HEAD"],
+        )
+        .decode()
+        .strip()
+    )
+
+
+def get_issue_or_pr_body(number: int) -> str:
```
Comment (osalpekar) [inline tools/testing/target_determination/heuristics/mentioned_in_pr.py:93]: Same q as above - should stuff like this belong in some general-purpose utils?

**Review by osalpekar — APPROVED** [review]
LGTM, left a few comments/questions

File: tools/testing/target_determination/heuristics/mentioned_in_pr.py (hunk 4 above)
Comment (clee2000) [reply tools/testing/target_determination/heuristics/mentioned_in_pr.py:67]: moved to utils file

File: tools/testing/target_determination/heuristics/mentioned_in_pr.py (hunk 3 above)
Comment (clee2000) [reply tools/testing/target_determination/heuristics/mentioned_in_pr.py:17]: edited comment to be more clear and have examples

**Pull request closed without merging**